packages/chain-adapters/src/cosmossdk/CosmosSdkBaseAdapter.ts

```diff
@@ -363,7 +363,7 @@ export abstract class CosmosSdkBaseAdapter<T extends CosmosSdkChainId> implement
     const { pubKey, accountNumber, wallet } = input
 
     const bip44Params = this.getBIP44Params({ accountNumber })
-    const address = await this.getAddress({ accountNumber, wallet, pubKey })
+    const address = await this.getAddress({ accountNumber, wallet, pubKey: pubKey as string })
```

Comment:
not a mistake, see `getAddress()` types update below

Reply:
Doing the check above would produce throws, see

https://github.com/shapeshift/web/blob/361180f2372e57fc99cfde2d5d9ea1247d59f750/src/context/TransactionsProvider/TransactionsProvider.tsx#L227-L232


I'd rather lie to the type system in this particular case, and keep the underlying type strict  - the reason for that is the type is as below, which covers *most* use-cases (either `checkLedgerAppOpenIfLedgerConnected`, *or* a defined `pubKey`, in which case we bypass on-device derivation, hence don't care about correct app being opened), but doesn't cover the rare case where `pubKey` is undefined, and `checkLedgerAppOpenIfLedgerConnected()` is also missing.

```typescript
(
  | {
      /**
       * An optional public key to be passed, which will bypass the HD wallet derivation
       * and instead use unchained to "derive" the address from the public key
       */
      pubKey: string
      checkLedgerAppOpenIfLedgerConnected?: never
    }
  | {
      /**
       * If no pubKey is passed, check if the Ledger device app is open before deriving the address on-device
       */
      checkLedgerAppOpenIfLedgerConnected: (chainId: ChainId) => Promise<void>
      pubKey?: never
    }
)
```

@0xApotheosis perhaps a comment would help here, WDYT?

Reply:
Disregard the above, this guy was the goer https://github.com/shapeshift/web/pull/7655#discussion_r1743143840: e08c8e4ce8